src/rgw/rgw_admin_remote.cc

```diff
@@ -0,0 +1,247 @@
+{
+  if (!opt_val) {
+    return;
+  }
+  add_param_str(params, name, *opt_val);
+}
+
+static inline void add_param_bool(std::vector<pair<string, string> >& params, const char *name, bool val)
+{
+  string v = (val ? "true" : "false");
+  add_param_str(params, name, v);
+}
+
+static inline void add_param_bool(std::vector<pair<string, string> >& params, const char *name, std::optional<bool> opt_val)
+{
+  if (!opt_val) {
+    return;
+  }
+  add_param_bool(params, name, *opt_val);
+}
+
+static inline void add_param_int(std::vector<pair<string, string> >& params, const char *name, int val)
+{
+  char buf[32];
+  snprintf(buf, sizeof(buf), "%d", val);
+  add_param_str(params, name, string(buf));
+}
+
+static inline void add_param_int(std::vector<pair<string, string> >& params, const char *name, std::optional<int> opt_val)
+{
+  if (!opt_val) {
+    return;
+  }
+  add_param_int(params, name, *opt_val);
+}
+
+
+int RGWRemoteAdminOp::create_user(const DoutPrefixProvider *dpp,
+                                  RGWRESTConn *conn,
+                                  const CreateUserParams& params,
+                                  optional_yield y)
+{
+  if (!conn) {
+    ldpp_dout(dpp, 1) << __func__ << "(): connection is not initialized" << dendl;
+    return -EINVAL;
+  }
+
+  std::vector<pair<string, string> > req_params;
+  add_param_str(req_params, "uid", params.user.id);
+  add_param_str(req_params, "tenant", params.user.tenant);
+  add_param_str(req_params, "display-name", params.display_name);
+  add_param_str(req_params, "email", params.email);
+  add_param_str(req_params, "access-key", params.access_key);
+  add_param_str(req_params, "secret-key", params.secret_key);
+  add_param_str(req_params, "key-type", params.key_type);
+  add_param_str(req_params, "user-caps", params.user_caps);
+  add_param_bool(req_params, "generate-key", params.generate_key);
+  add_param_bool(req_params, "suspended", params.suspended);
+  add_param_int(req_params, "max-buckets", params.max_buckets);
+  add_param_int(req_params, "system", params.system);
+  add_param_bool(req_params, "exclusive", params.exclusive);
+  add_param_str(req_params, "op-mask", params.op_mask);
+  add_param_str(req_params, "default-placement", params.default_placement);
+  add_param_str(req_params, "placement-tags", params.placement_tags);
+
+  bufferlist outbl;
+  int r = send_remote(dpp,
+                      conn,
+                      "PUT",
+                      "/admin/user",
+                      req_params,
+                      &outbl,
+                      y);
+  if (r < 0) {
+    ldpp_dout(dpp, 5) << "remote request failed: r=" << r << dendl;
+    return r;
+  }
+  return 0;
+}
+
+int RGWRemoteAdminOp::create_user(const DoutPrefixProvider *dpp,
+                                  const CreateUserParams& params,
+                                  optional_yield y)
```

Comment:
Without this, the creds creation command for new zone needs to happen on the master

Reply:
is this creating a separate user for each zone? why not share a single system user?

Reply:
I'm not sure that sharing the same user would be the best practice security wise

Reply:
what kind of vulnerability would this prevent? metadata sync will replicate these users to every zone, and radosgw-admin can read all of their keys anyway

Reply:
ok, i think this needs some more elaboration as a security model, like an extra command to destroy the user we created with `create zone-creds` and documentation explaining the process

since the token includes an endpoint, it can be invalidated if the master zone endpoints change, or after failover to a new master zone. in that case, the admin would need to create another token, and know to clean up the user we created for the previous token with something like `destroy zone-creds`

from the description of `ceph rgw realm bootstrap`,
> Command returns a realm token that allows new zones to easily join this realm

so zones created with this realm token would follow the 'shared system user' model that i prefer. i would also suggest another command like `ceph rgw realm token` that can regenerate that token in case the endpoints changed, or the admin lost the token from bootstrap

the `create zone-creds` command does require the master zone, so the rgw_admin_remote.cc bits still shouldn't be necessary
```python
        if period.master_zone != zone_id:
            return (-errno.EINVAL, '', 'Command needs to run on master zone')
```

Reply:
@cbodley added `ceph rgw zone-creds remove` command, removed the `rgw_admin_remote.cc` stuff per your suggestions.